shelley/chain-and-ledger/executable-spec/src/TxData.hs

```diff
@@ -384,22 +474,60 @@ instance (Crypto crypto) =>
     (b :: Word64) <- fromCBOR
     pure $ TxIn a (fromInteger $ toInteger b)
 
+
+
 instance
   (Typeable crypto, Crypto crypto)
-  => ToCBOR (TxOut crypto)
+  => ToCBOR (TxInTx crypto)
  where
-  toCBOR (TxOut addr coin) =
-    encodeListLen 2
-      <> toCBOR addr
-      <> toCBOR coin
+  toCBOR = \case
+    TxInVK txin isfee ->
+      encodeListLen 3
+       <> toCBOR (0 :: Word8)
+       <> toCBOR txin
+       <> toCBOR isfee
+
+    TxInScr txin dh ->
+      encodeListLen 3
+       <> toCBOR (1 :: Word8)
+       <> toCBOR txin
+       <> toCBOR dh
+
+instance (Crypto crypto) =>
+  FromCBOR (TxInTx crypto) where
+  fromCBOR = do
+    n <- decodeListLen
+    decodeWord >>= \case
+      0 -> do
+        matchSize "TxInVK" 2 n
+        a <- fromCBOR
+        b <- fromCBOR
+        pure $ TxInVK a b
+      1 -> do
+        matchSize "TxInScr" 2 n
+        a <- fromCBOR
+        (b ) <- fromCBOR
+        pure $ TxInScr a (DataHash b)
+      k -> invalidKey k
+
 
 instance (Crypto crypto) =>
   FromCBOR (TxOut crypto) where
   fromCBOR = do
-    enforceSize "TxOut" 2
-    a <- fromCBOR
-    (b :: Word64) <- fromCBOR
-    pure $ TxOut a (Coin $ toInteger b)
+    n <- decodeListLen
+    decodeWord >>= \case
+      0 -> do
+        matchSize "TxOutVK" 2 n
+        a <- fromCBOR
+        b <- fromCBOR
+        pure $ TxOutVK a b
+      1 -> do
+        matchSize "TxOutVK" 3 n
```

Comment:
`TxOutScr` ?

Reply:
`TxOutScr` is an output that has the hash of term of type `Data` in it (along with the usual value and address), which is needed for script validation